warp10/src/main/java/io/warp10/script/functions/TOLONGBYTES.java

```diff
@@ -49,13 +50,47 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
         } else {
           stack.push(Longs.toByteArray(((Long) o).longValue()));
         }
+      } else if (o instanceof List && nbBytes > 0 && nbBytes <= 8) {
+        try {
+          ByteBuffer bytes = ByteBuffer.allocate(nbBytes * ((List) o).size());
+          if (8 == nbBytes) {
+            for (Object el : (List) o) {
+              bytes.putLong((Long) el);
+            }
+          } else if (4 == nbBytes) {
+            for (Object el : (List) o) {
+              bytes.putInt(((Long) el).intValue());
+            }
+          } else if (2 == nbBytes) {
+            for (Object el : (List) o) {
+              bytes.putShort(((Long) el).shortValue());
+            }
+          } else if (1 == nbBytes) {
+            for (Object el : (List) o) {
+              bytes.put(((Long) el).byteValue());
+            }
+          } else {
+            // weird length are not optimized
+            long v;
+            for (Object el : (List) o) {
+              v = ((Long) el).longValue() << (8 - nbBytes) * 8;
+              for (int i = nbBytes - 1; i >= 0; i--) {
```

Comment:
since you are not using i, why not use an increasing index? It would make things clearer